fast64_internal/oot/oot_utility.py

```diff
@@ -586,3 +586,57 @@ def oot_utility_register():
 def oot_utility_unregister():
 	for cls in reversed(oot_utility_classes):
 		unregister_class(cls)
+
+def getIDFromKey(key, root):
+	'''Returns the actor/object ID using the key'''
+	if not (key == 'Custom'):
+		for node in root:
+			dataKey = node.get('Key')
+			if dataKey is not None and dataKey == key:
+				return node.get('ID')
+	else:
+		return key
+	return None
+
+def addMissingObjectsToList(roomObj, room, actorRoot, headerIndex, csHeaderIndex):
+	'''Adds missing objects to the object list'''
+	if len(room.actorList) > 0:
+		for actor in room.actorList:
+			for actorNode in actorRoot:
+				if actorNode.get('ID') == actor.actorID and not (actorNode.get('Key') == 'player'):
+					dataObj = actorNode.get('ObjectKey')
+					if dataObj is not None:
+						for obj in dataObj.split(','):
+							if not (obj in room.objectList) and not (obj.startswith('obj_gameplay')):
+								room.objectList.append(obj)
+								addMissingObjectToUI(roomObj, headerIndex, obj, csHeaderIndex)
+
+def addMissingObjectToUI(roomObj, headerIndex, objectID, csHeaderIndex):
+	'''Add the missing object to the room empty object OoT object list'''
+	if roomObj is not None:
+		if headerIndex == 0:
+			roomProp = roomObj.ootRoomHeader
+		elif headerIndex == 1:
+			roomProp = roomObj.ootAlternateRoomHeaders.childNightHeader
+		elif headerIndex == 2:
+			roomProp = roomObj.ootAlternateRoomHeaders.adultDayHeader
+		elif headerIndex == 3:
+			roomProp = roomObj.ootAlternateRoomHeaders.adultNightHeader
+		elif csHeaderIndex is not None:
+			roomProp = roomObj.ootAlternateRoomHeaders.cutsceneHeaders[csHeaderIndex]
+		if roomProp is not None:
+			collection = getCollection(roomObj.name, "Object", headerIndex)
+			collection.add()
+			collection.move(len(collection)-1, (headerIndex + 1))
+			roomProp.objectList[len(roomProp.objectList) - 1].objectID = objectID
```

Comment:
`collection` / `roomProp.objectList` ?

what does move() do here? (or in general)

```suggestion
			collection.move(len(collection)-1, (headerIndex + 1))
			roomProp.objectList[-1].objectID = objectID
```

Reply:
I never worked with "collections" before so I just did exactly what the "add" button does when you're adding objects

So if I change this I'll change the other code too to be consistent

Reply:
collection methods have no api page so it's very cool to use YEP

Reply:
no documentation, love it YEP